### PR TITLE
feat: add dual region bucket support and sample

### LINKED ISSUE #739: Add Support for Dual Region Buckets
**FR:** Support dual region buckets.

The method for creating a bucket should accept a string with 2 regions concatenated with a '+' (e.g. `US-EAST1+US-WEST1`).

**System tests:**
- Verify a dual region bucket with specified regions can be created
- Verify a dual region bucket's `location` metadata is the specified dual region provided
- Verify a dual region bucket's `locationType` metadata is `dual-region`

[Internal] For details and coordination: [go/gcs-dpe-dual-region-milestones](http://go/gcs-dpe-dual-region-milestones)

### PR DESCRIPTION
**Pull request by cojenco**
Add dual region bucket support and sample
- method `client.create_bucket()` accepts a dual-region `location` as a string with 2 regions concatenated with a '+' (e.g. `"US-EAST1+US-WEST1"`)
- system test verifies a dual region bucket with specified regions can be created with correct location and location_type
- create dual region bucket sample 

Fixes #739 🦕

### CODE REVIEW

**Review by danielbankhead — APPROVED** [review]
LGTM, one nit - we'll want to update references from 'dual region' to 'dual-region'

**Comment by cojenco** [comment]
Update - Python samples projects have been added to the allowlist (internal cl/438869140). Awaiting for deployment in the next 2 weeks.

**Comment by cojenco** [comment]
> LGTM, one nit - we'll want to update references from 'dual region' to 'dual-region'

@danielbankhead thanks for the review! Do you mean updating the docstrings in client.py?

**Comment by danielbankhead** [comment]
Yep, correct

**Comment by cojenco** [comment]
> Yep, correct

Updated the docstrings and the docs link as discussed in java thread.

**Comment by cojenco** [comment]
Note(04.06.2022): Dual-region storage is currently a Pre-GA feature https://cloud.google.com/storage/docs/dual-regions

File: google/cloud/storage/client.py [hunk 1]
```diff
@@ -875,8 +875,9 @@ def create_bucket(
                 made via created bucket.
             location (str):
                 (Optional) The location of the bucket. If not passed,
-                the default location, US, will be used. See
-                https://cloud.google.com/storage/docs/bucket-locations
+                the default location, US, will be used. If specifying a dual-region,
```
Comment (andrewsg) [inline google/cloud/storage/client.py:878]: May I suggest the following:

(Optional) The location of the bucket. If not passed, the default location, 'US', will be used. If specifying a dual-region location, the regions can be specified by joining them with a plus sign, for example 'US-CENTRAL1+US-WEST1'. See:
                https://cloud.google.com/storage/docs/locations

**Pull request merged**